modules/library/jdbc/src/main/java/org/geotools/jdbc/MetadataTablePrimaryKeyFinder.java

```diff
@@ -112,17 +113,26 @@ public PrimaryKey getPrimaryKey(JDBCDataStore store, String schema, String table
                 synchronized (this) {
                     if (metadataTableExists == null) {
                         try {
+                            // This is a database-agnostic way to check if the table exists without
+                            // resulting in error messages. It also checks for case sensitivity.
+                            tablesRs =
+                                    cx.getMetaData()
+                                            .getTables(
+                                                    null,
+                                                    metadataSchema,
+                                                    "%",
+                                                    new String[] {"TABLE", "VIEW"});
 
-                            StringBuffer sb = new StringBuffer();
-                            sb.append("SELECT * FROM ");
-                            if (metadataSchema != null) {
-                                store.getSQLDialect().encodeSchemaName(metadataSchema, sb);
-                                sb.append(".");
+                            metadataTableExists = false;
+
+                            while (tablesRs.next()) {
+                                String tblName = tablesRs.getString(3);
+                                if (tblName.toLowerCase().equals(tableName.toLowerCase())) {
```

Comment:
I would keep this case insensitive test, not sure databases are returning the table name in a predicatable case (see also my other comments).

Reply:
Done.